File: src/main/java/openmedstack/events/BaseEvent.java
package openmedstack.events;

import openmedstack.*;
import java.time.Instant;

public abstract class BaseEvent implements ICorrelate {
    private final String _source;
    private final Instant _timeStamp;
    private final String _correlationId;

    protected BaseEvent(String source, Instant timeStamp) {
        this(source, timeStamp, null);
    }


    protected BaseEvent(String source, Instant timeStamp, String correlationId) {
        if (timeStamp == Instant.MIN) {
            throw new IllegalArgumentException();
        }

        _source = source;
        _timeStamp = timeStamp;
        _correlationId = correlationId;
    }

    public String getSource(){
        return _source;
    }

    public Instant getTimestamp(){
        return _timeStamp;
    }

    @Override
    public String getCorrelationId(){
        return _correlationId;
    }
}

File: src/main/kotlin/org/openmedstack/events/BaseEvent.kt
package org.openmedstack.events

import java.time.Instant
import org.openmedstack.ICorrelate
import java.lang.IllegalArgumentException
import org.openmedstack.events.BaseEvent
import org.openmedstack.MessageHeaders
import java.util.concurrent.CompletableFuture
import java.util.HashMap
import org.openmedstack.events.IHandleEvents
import java.lang.AutoCloseable
import org.openmedstack.commands.DomainCommand
import org.openmedstack.commands.CommandResponse
import org.openmedstack.commands.IHandleCommands
import org.openmedstack.DeploymentConfiguration
import org.openmedstack.Chassis
import kotlin.Throws
import java.lang.NullPointerException
import org.openmedstack.ILookupServices
import org.openmedstack.IValidateTokens

abstract class BaseEvent protected constructor(source: String, timeStamp: Instant, correlationId: String? = null) : ICorrelate {
    val source: String
    val timestamp: Instant
    override val correlationId: String?

    init {
        require(!(timeStamp === Instant.MIN))
        this.source = source
        timestamp = timeStamp
        this.correlationId = correlationId
    }
}